VideoWeb/VideoWeb/ClientApp/src/app/waiting-space/select-hearing-layout/select-hearing-layout.component.ts

```diff
@@ -25,9 +26,22 @@ export class SelectHearingLayoutComponent implements OnInit {
 
         (<any>window).GOVUKFrontend.initAll();
         headingElement.onclick = e => this.setAccordionText(e);
+        const sectionHeadingElement = document.getElementsByClassName('govuk-accordion__section-button').item(0) as HTMLButtonElement;
+        sectionHeadingElement.onclick = e => this.setAccordionText(e);
         this.accordionOpenAllElement = document.getElementsByClassName('govuk-accordion__open-all').item(0) as HTMLButtonElement;
         this.accordionOpenAllElement.onclick = e => this.setAccordionText(e);
         this.setAccordionText({} as MouseEvent);
+
+        this.translateService.onLangChange.subscribe(event => {
```

Comment:
Implement ngOnDestroy and do it in there.

Reply:
I will add ngondestroy, however I thought this would be a default behaviour, as in when this component gets unmounted shouldnt it all get removed/unsubscribed?

Reply:
It is  a complicated topic. It depends on the service implementation I believe. It is good practice to do it. In this case though the service will be alive for the entire lifetime of the app so it is safe to assume that the observable will never complete. Our component wont automatically unsubscribe from the observable. https://stackoverflow.com/questions/38008334/angular-rxjs-when-should-i-unsubscribe-from-subscription